ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/OptionsDropdown.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ReactNode } from 'react';
+import { Dropdown, DropdownToggle, Button, FocusTrap } from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons';
+
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+
+export type OptionsDropdownProps = {
+    children: ReactNode;
+    toggleId: string;
+};
+
+/**
+ * Component that uses a PatternFly `Dropdown` with the contents wrapped in a `FocusTrap` to
+ * enhance accessibility for interactive elements. Upon opening the dropdown, the browser will focus
+ * on the container's close button and only allow tab navigation within the container.
+ */
+function OptionsDropdown({ children, toggleId }: OptionsDropdownProps) {
```

Comment:
I have a general question about focus trapping here. So does this mean that PatternFly uses focus trapping under the hood for components like the Modal and Popover but not for the Dropdown? If this is something that the Dropdown didn't have intentionally, is there a reason for that? If they didn't, and maybe they should have, is it worth bringing it up for consideration on their part?

Reply:
Because other occurrences have `dropdownItems` prop, in which you do not tab between items; but these have form as child, in which you tab between fields, correct?

Reply:
Having said that, System Health diagnostic bundle form is similar to dashboard:
* current classic-PatternFly hybrid: https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/SystemHealth/Components/DiagnosticBundleDialogBox.tsx#L146-L211
* future PatternFly is child of `Popover` element: https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/SystemHealth/PatternFly/Components/GenerateDiagnosticBundle.tsx#L102-L135

Reply:
I think this is because the main use case for the Dropdown seems to be a list of items/buttons as shown in all of the demos like Mark mentioned, but I'm passing a form directly as child props. The [PF docs](https://www.patternfly.org/v4/components/dropdown#dropdown) say you can pass a `ReactNode` of "Anything which can be rendered in a dropdown". I can mention it to the PF team and see if this was an intentional omission, or if it is just a rare enough use case that they haven't implemented it.

Let me take a look at doing this with a `Popover` as well, and see if that implementation is cleaner and keeps the same functionality.